### ISSUE TITLE
Documentation Examples

### ISSUE THREAD

**Issue by Thetwam** [ucfopen/canvasapi#147]
As a stopgap measure until we complete #9, we should provide code examples for common use cases.

Some important examples:

- Create a new user
- Create a new course
- Enroll user(s) in a course
- Create/edit an assignment
- Get all users in a course

There's plenty of other things to add, of course. Add more as needed.

We'll also need to figure out the best way to organize all these examples. For now, a single page in the docs should suffice.

**Comment by Thetwam** [comment]
@jessemcbride (and anybody else who is interested), could you take a look at [my branch](https://github.com/Thetwam/canvasapi/tree/issue/147-docs-examples) and see if this seems like a good pattern to continue with?

**Comment by Thetwam** [comment]
@jessemcbride things to work on:

Accounts:

- [ ] Creating a new user
- [ ] List all courses

Users
- [ ] Getting a user by SIS ID
- [ ] Getting a user by their Canvas ID
- [ ] Edit a user
- [ ] Get page views

Logins

- [ ] List user logins
- [ ] Adding a user login
- [ ] Updating a user login

**Comment by Thetwam** [comment]
Work continues on this. Will likely be an ever-evolving issue.

**Comment by mboldin** [comment]
Any examples of how to set grades for assignments?
If someone can give me a hint I will make an example
I see how  the assignment class  .set_attributes( {x.grade: X}  )  works but not how to POST grades
I imagine I need to post / upload the students assignment object with changed attributes

**Comment by Thetwam** [comment]
@mboldin You should be able to grade an assignment via [`Submission.edit()`](http://canvasapi.readthedocs.io/en/latest/submission-ref.html#canvasapi.submission.Submission.edit) using the `submission[posted_grade]` parameter [as described in the Canvas docs](https://canvas.instructure.com/doc/api/submissions.html#method.submissions_api.update).

**Comment by mboldin** [comment]
Thanks. The code below works. The trick is you do not use the grade parameter
   i.e., NOT  edit(submission={'grade': new_grade})
And it is easy to iterate over student ids  or use .get_submissions() to loop over submissions
thanks to other Python code in CANVASAPI

```
## Pick course, assignment and student
course = canvas.get_course( -- course number --- )
a = course.get_assignment( -- assignment id --)
x = -- student ID  --

## Modify assignment grade for one student 
print (a.name, '--', a.created_at[:10], a.due_at[:10], a.updated_at)
s1 = b.get_submission(x)
print(x, s1.id, 'Current grade:', s1.grade)
new_grade = 10
s1.edit(submission={'posted_grade': new_grade})

##Check result
s2 = b.get_submission(x)
print(x, s2.id, 'Modified:', s2.grade)

```

**Comment by Thetwam** [comment]
Yeah, per the [Canvas Docs](https://canvas.instructure.com/doc/api/submissions.html#method.submissions_api.update), `submission[posted_grade]` will update both the `score` and `grade` attributes.

Also, the object should update in-place, so in your example `s1` should already have the modified grade. If not, let us know!

**Comment by elec3647** [comment]
I'd love to help! Is anyone tackling creating and managing assignments? I also found some issues with uploading files.

**Comment by Thetwam** [comment]
Hey @elec3647!

We already have [examples for creating and editing assignments](http://canvasapi.readthedocs.io/en/latest/examples.html#assignments). If you have suggestions to improve those examples, or have any other examples to add, please feel free to submit a pull request!

Thanks 😄

**Comment by jessemcbride** [comment]
Add examples in the docs for CalendarEvent.edit() (see #239).

**Comment by dave-doty** [comment]
Can I make a request related to this issue? 

In addition to the examples, for at least a few examples can you add a bit more "tutorial" explanation on how to map from the Canvas API (https://canvas.instructure.com/doc/api/) to the equivalent in the `canvasapi` Python package?

Currently, there's a few examples related to the current issue, but no explanation of how one would discover them by looking at the Canvas API documentation.

My way of determining how objects in the Python package are structured (what fields they contain) is to run a debugger with a breakpoint, then inspect the object in the debugger.

I realize that having full documentation of all the fields and possible keyword args is an open issue (#9), but in the meantime it would help if the examples explained the reasoning used to go from the Canvas API docs at https://canvas.instructure.com/doc/api/ to their equivalent in the Python package, e.g., linking to the relavant part of the Canvas API and explaining how the concepts presented there show up in the Python package.
